js/app.js

```diff
@@ -1,89 +1,120 @@
+function addNewInvitee() {
+	const newInvitee = new Invitee(nextId, addInviteeInput.value);
+	invitees = [...invitees, newInvitee];
+	addInviteeInput.value = ``;
+	nextId++
+}
 
-	function submitForm() {
-		const newLi = newElement(invitedUl, 'li');
-		const newSpan = text(newLi, `span`, formInput.value);
-		const newInput = type(newLi, `input`, 'text');
-		newInput.value = formInput.value;
-		newInput.classList.add(`invisible`);
-		const label = text(newLi, `label`, `Confirmed`);
-		label.setAttribute(`for`, `confirmed`);
-		const checkbox = type(label, 'input', `checkbox`);
-		checkbox.setAttribute(`name`, `confirmed`);
-		const edit = text(newLi, 'button', 'Edit');
-		const remove = text(newLi, 'button', `Remove`);
-	}
+function indexOfId(id) {
+	const index = invitees.findIndex(invitee => invitee.id === id);
+	return index;
+}
 
+function editOrRemoveInvitee(button) {
+	const buttonName = button.textContent.toLowerCase();
+	const inviteeId = parseInt(button.dataset[buttonName]);
+	const inviteeIndex = indexOfId(inviteeId);
+	const invitee = invitedUl.querySelector(`#invitee-${inviteeId}`);
+	const buttonActions = {
+		remove: () => {
+			invitees = [
+				...invitees.slice(0,inviteeIndex),
+				...invitees.slice(inviteeIndex + 1)
+			];
+			renderAllInvitees();
+		},
+		edit: () => {
+			invitee.querySelector('[data-name]').classList.add('invisible');
+			invitee.querySelector('[data-input]').classList.remove('invisible');
+			button.textContent = `Save`;
+		},
+		save: () => {
+			const newName = invitee.querySelector('[data-input]').value;
+			invitees[inviteeIndex].changeName(newName);
+			renderAllInvitees();
+		}
+	};
+	buttonActions[buttonName]();
+}
 
-	form.addEventListener(`submit`, (e) => {
-		e.preventDefault();
-		submitForm();
-		formInput.value = ``;
+/* -----------------
+ Rendering Function
+------------------ */
+
+function renderAllInvitees() {
+	while (invitedUl.firstChild) {
+		invitedUl.removeChild(invitedUl.firstChild);
+	}
+	const isChecked = hideNotRespondedCheckbox.checked;
+	let inviteesToDisplay;
+	if (isChecked) {
+		inviteesToDisplay = invitees.filter(invitee => invitee.confirmed);
+	} else{
+		inviteesToDisplay = invitees;
+	}
+	const docFrag = document.createDocumentFragment();
+	inviteesToDisplay.forEach(invitee => {
+		const inviteeTile = emptyInviteeTile.cloneNode(true);
+		inviteeTile.setAttribute('id', `invitee-${invitee.id}`);
+		inviteeTile.querySelector('[data-name]').textContent = invitee.name;
+		inviteeTile.querySelector('[data-input]').value = invitee.name;
+		inviteeTile.querySelector('[data-input]').dataset.input = invitee.id;
+		inviteeTile.querySelector('[data-edit]').dataset.edit = invitee.id;
+		inviteeTile.querySelector('[data-save]').dataset.save = invitee.id;
+		inviteeTile.querySelector('[data-remove]').dataset.remove = invitee.id;
+		inviteeTile.querySelector('[data-confirmed]').dataset.confirmed = invitee.id;
+		inviteeTile.querySelector('[name="confirmed"]').checked = invitee.confirmed;
```

Comment:
To mi się nie podoba. Masz tutaj bardzo dużo atrybutów ustawianych. Aż pięć z nich ma wartość `invitee.id`.
Zastanów się czy nie możesz tego czasem inaczej rozwiązać (poza `value` inputa, bo ten pewnie musi zostać tak jak jest) - np. nadać atrybut `data-invite-container` o wartości `invitee.id` i następnie nadać `click` event listener na ten container i w zależności od tego jaki element został kliknięty egzekwować odpowiedni kod. Wtedy będziesz miał `invitee.id` w jednym miejscu i po prostu będziesz odpowiednio wykorzystywał w zależności od tego co zostało kliknięte